### PR TITLE
[CDAP-14180] Copy common-http to cdap-common module

### PR DESCRIPTION
**Pull request by anew**
this is to be able make changes easier as we clean up test code. we may move it back out later.

### CODE REVIEW

**Review by chtyim — APPROVED** [review]
LGTM

**Comment by anew** [comment]
Build passed: https://builds.cask.co/browse/CDAP-DUT6622/latest

File: cdap-tms/pom.xml [hunk 1]
```diff
@@ -42,10 +42,6 @@
       <version>${project.version}</version>
     </dependency>
 
-    <dependency>
```
Comment (anwar6953) [inline cdap-tms/pom.xml:45]: Do you also want to remove it from https://github.com/caskdata/cdap/blob/develop/cdap-examples/Purchase/pom.xml#L45-L48?

File: cdap-client/src/main/java/co/cask/cdap/client/util/RESTClient.java [hunk 2]
```diff
@@ -19,12 +19,12 @@
 import co.cask.cdap.client.config.ClientConfig;
 import co.cask.cdap.client.exception.DisconnectedException;
 import co.cask.cdap.common.UnauthenticatedException;
+import co.cask.cdap.common.http.HttpMethod;
```
Comment (anwar6953) [inline cdap-client/src/main/java/co/cask/cdap/client/util/RESTClient.java:22]: Any other projects that use this class (RESTClient) would have compilation failures.

File: cdap-tms/pom.xml (hunk 1 above)
Comment (anwar6953) [reply cdap-tms/pom.xml:45]: There's also a dependency on the common-http module in other places, such as in the `cdap-clients` module.

File: cdap-client/src/main/java/co/cask/cdap/client/util/RESTClient.java (hunk 2 above)
Comment (anwar6953) [reply cdap-client/src/main/java/co/cask/cdap/client/util/RESTClient.java:22]: Fixed one instance in: https://github.com/caskdata/cdap-integration-tests/pull/954

File: cdap-tms/pom.xml (hunk 1 above)
Comment (anew) [reply cdap-tms/pom.xml:45]: I think that apps should depend on common.http as they used to, and our examples are apps. I am trying to only change the internal usages of http within CDAP.

File: cdap-client/src/main/java/co/cask/cdap/client/util/RESTClient.java (hunk 2 above)
Comment (anew) [reply cdap-client/src/main/java/co/cask/cdap/client/util/RESTClient.java:22]: good catch. I will send another PR that will restore the compatibility

**Pull request merged**